pkg/networkservice/common/mechanisms/recvfd/server_test.go

```diff
@@ -0,0 +1,283 @@
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		n.t.Fatal("No peer in context")
+	}
+
+	transceiver, ok := p.Addr.(grpcfd.FDTransceiver)
+	assert.True(n.t, ok)
+
+	p.Addr = &wrapTransceiver{
+		FDTransceiver: transceiver,
+		Addr:          p.Addr,
+		onFileClosed:  n.onFileClosed,
+	}
+
+	return next.Server(ctx).Request(ctx, request)
+}
+
+func (w *wrapTransceiver) RecvFileByURL(urlStr string) (<-chan *os.File, error) {
+	res, err := w.FDTransceiver.RecvFileByURL(urlStr)
+	if err != nil {
+		return nil, err
+	}
+
+	var chanRes = make(chan *os.File)
+	go func() {
+		for f := range res {
+			runtime.SetFinalizer(f, func(file *os.File) {
+				onFileClosedFunc, ok := w.onFileClosed[urlStr]
+				if ok {
+					onFileClosedFunc()
+				}
+			})
+			chanRes <- f
+		}
+	}()
+
+	return chanRes, nil
+}
+
+func createFile(fileName string, t *testing.T) (inodeURLStr string, fileClosedContext context.Context, cancelFunc func()) {
+	f, err := os.Create(fileName)
+	require.NoError(t, err, "Failed to create and open a file: %v", err)
+
+	info, err := f.Stat()
+	assert.NoError(t, err)
+
+	stat, ok := info.Sys().(*syscall.Stat_t)
+	assert.True(t, ok)
+
+	err = f.Close()
+	require.NoError(t, err, "Failed to close file: %v", err)
+
+	fileClosedContext, cancelFunc = context.WithCancel(context.Background())
+	inodeURLStr = fmt.Sprintf("inode://%d/%d", stat.Dev, stat.Ino)
+
+	return
+}
+
+func createServerAndClient(ctx context.Context, t *testing.T, testServerChain *networkservice.NetworkServiceServer, serveURL *url.URL) (testClient networkservice.NetworkServiceClient) {
+	var grpcServer = grpc.NewServer(grpc.Creds(grpcfd.TransportCredentials(insecure.NewCredentials())))
+	networkservice.RegisterNetworkServiceServer(grpcServer, *testServerChain)
+
+	var errCh = grpcutils.ListenAndServe(ctx, serveURL, grpcServer)
+
+	select {
+	case e := <-errCh:
+		assert.Failf(t, "Server failed to start: %v", e.Error())
+	default:
+	}
+
+	testClient = client.NewClient(
+		ctx,
+		client.WithClientURL(sandbox.CloneURL(serveURL)),
+		client.WithDialOptions(grpc.WithTransportCredentials(
+			grpcfd.TransportCredentials(insecure.NewCredentials())),
+		),
+		client.WithDialTimeout(time.Second),
+		client.WithoutRefresh(),
+		client.WithAdditionalFunctionality(sendfd.NewClient()))
+	return
+}
+
+func TestRecvfdClosesSingleFile(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	dir, err := ioutil.TempDir(os.TempDir(), t.Name())
+	require.NoError(t, err)
+	defer func() {
+		_ = os.RemoveAll(dir)
+	}()
+
+	s, err := os.Create(path.Join(dir, "test.sock"))
+	require.NoError(t, err)
+
+	var testFileName = path.Join(dir, t.Name()+".test")
+
+	inodeURLStr, fileClosedContext, cancelFunc := createFile(testFileName, t)
+
+	serveURL := &url.URL{Scheme: "unix", Path: s.Name(), Host: "0.0.0.0:5000"}
```

Comment:
Could you explain why do wee need a **Host** for unix?

Reply:
It's not required, I've removed it